### PR TITLE
SECURESIGN-1362 | Trillian failing to start after upgrade

### PR DESCRIPTION
**Pull request by JasonPowr**
The main cause for trillian not starting after upgrade was because the existing service was missing the 'service.beta.openshift.io/serving-cert-secret-name' annotation meaning the secrets for TLS communication were not being created.

### CODE REVIEW

File: internal/controller/common/action/base_action.go [hunk 1]
```diff
@@ -125,6 +140,19 @@ func (action *BaseAction) Ensure(ctx context.Context, obj client2.Object) (bool,
 		}
 	}
 
+	// Normalize both current and desired annotations
+	currentAnnotations := normalizeAnnotations(annotations.FilterManaged(currentObj.GetAnnotations()))
+	desiredAnnotations := normalizeAnnotations(annotations.FilterManaged(obj.GetAnnotations()))
+	// Compare annotations, and update if they are different
+	if !reflect.DeepEqual(currentAnnotations, desiredAnnotations) {
```
[the remark this answers is not in the document]
Comment (JasonPowr) [reply internal/controller/common/action/base_action.go:147]: Ok Ill take a look

[the remark this answers is not in the document]
Comment (osmman) [reply internal/controller/common/action/base_action.go:147]: Is it caused by same problem like https://github.com/securesign/secure-sign-operator/pull/613?

[the remark this answers is not in the document]
Comment (JasonPowr) [reply internal/controller/common/action/base_action.go:147]: Its similar, the main problem being that we only check and compare the spec of objects (I believe).

In this particular case, if the environment is openshift https://github.com/securesign/secure-sign-operator/blob/main/internal/controller/trillian/actions/db/svc.go#L48 we annotate the service to create the TLS secrets needed for TLS communication, this is fine when creating the service object, but on upgrade the current implementation doesnt reconcile and add the annotation leading to the error `MountVolume.SetUp failed for volume "tls-cert" : secret "securesign-sample-trillian-db-tls" not found `

**Comment by JasonPowr** [comment]
closing in favour off https://github.com/securesign/secure-sign-operator/pull/613

**Pull request closed without merging**